core/misc/collapse.js

```diff
@@ -15,19 +15,13 @@ Backdrop.toggleFieldset = function (fieldset) {
       duration: 'fast',
       easing: 'linear',
       complete: function () {
-        if (insideDialog === false) {
-          Backdrop.collapseScrollIntoView(fieldset);
-        }
         $fieldset.trigger({ type: 'collapsed', value: false });
         $(window).triggerHandler('resize');
         Backdrop.optimizedResize.trigger();
-        fieldset.animating = false;
-      },
-      step: function () {
-        // Scroll the fieldset into view unless inside a dialog.
         if (insideDialog === false) {
           Backdrop.collapseScrollIntoView(fieldset);
```

Comment:
@indigoxela Do we need this `collapseScrollIntoView` here?

Reply:
Yes, sure, the function Backdrop.collapseScrollIntoView checks if the expanded fieldset would slip outside the viewport and - depending on the size - scrolls into view either aligned on top (if the expanded fieldset is larger than the viewport) or on bottom, if it's smaller.

The diff looks a bit confusing, because the "step" is gone. Probably it's easier to view the full changed file to understand what it does.